docs/architecture/gameUtil.md

```diff
@@ -0,0 +1,52 @@
+# GameUtil
+
+The [GameUtil](../../src/utils/gameUtil.ts) orchestrates
+loading the [GameState](../data/gameState.md),
+aggregating the [CommandDescription](../data/commandDescription.md),
+[PlayerStateDescription](../data/playerStateDescription.md)
+and available [Command](../data/command.md)s,
+and saving the new game state.
+
+It interacts with the [GameStateUtil](./gameStateUtil.md) to save and load the GameState,
+the [RoomUtil](./roomUtil.md) to run the Command and retrieve
+the next [RoomHandler](./roomHandler.md), and the RoomHandler to get the PlayerStateDescription and available Commands.
+
+## Run
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant A as Actor
+  participant GU as GameUtil
+  participant RU as RoomUtil
+  participant RH1 as RoomHandler<T1>
+  participant RH2 as RoomHandler<T1 | T2>
+  participant GS as GameStateUtil
+
+  note over RH1, RH2: TX extends RoomType <br> T1 can equal T2
+
+  A ->> GU: run(command: string)
+
+  GU ->> GS: load()
+  GS ->> GU: gameState: GameState
+
+  GU ->> RU: getRoomHandlerByGameState(gameState)
+  RU ->> GU: roomHandlerA: RoomHandler<T1>
+
+  GU ->> RH1: run(gameState.roomState, command)
+  RH1 ->> GU: commandDescription: CommandDescription <br> roomState': RoomState<T1> | null
```

Comment:
Looking at the `null`. Will some commands not have a description? (I'm not sure what this line means)

Reply:
That's what it means. It's mainly for the default command handler which doesn't have a description. RoomHandlers can override the default command handler to change the room state, but they also wouldn't want to have a command description